x-pack/plugins/triggers_actions_ui/public/application/sections/actions_connectors_list/components/actions_connectors_list.test.tsx

```diff
@@ -76,95 +75,105 @@ describe('actions_connectors_list', () => {
           show: true,
         },
       };
-      wrapper = mountWithIntl(<ActionsConnectorsList />);
-
-      // Wait for active space to resolve before requesting the component to update
-      await act(async () => {
-        await nextTick();
-        wrapper.update();
-      });
-    }
+    };
 
     it('renders empty prompt', async () => {
       await setup();
-      expect(
-        wrapper.find('[data-test-subj="createFirstConnectorEmptyPrompt"]').find('EuiEmptyPrompt')
-      ).toHaveLength(1);
-      expect(
-        wrapper.find('[data-test-subj="createFirstActionButton"]').find('EuiButton')
-      ).toHaveLength(1);
+      render(
+        <IntlProvider>
+          <ActionsConnectorsList
+            setAddFlyoutVisibility={() => {}}
+            loadActions={async () => {}}
+            editItem={() => {}}
+            isLoadingActions={false}
+            actions={[]}
+            setActions={() => {}}
+          />
+        </IntlProvider>
+      );
+
+      expect(await screen.findByTestId('createFirstConnectorEmptyPrompt')).toBeInTheDocument();
+      expect(await screen.findByTestId('createFirstActionButton')).toBeInTheDocument();
     });
 
     test('if click create button should render CreateConnectorFlyout', async () => {
       await setup();
-      wrapper.find('[data-test-subj="createFirstActionButton"]').last().simulate('click');
-      expect(wrapper.find('[data-test-subj="create-connector-flyout"]').exists()).toBeTruthy();
+      const setAddFlyoutVisibility = jest.fn();
+      render(
+        <IntlProvider>
+          <ActionsConnectorsList
+            setAddFlyoutVisibility={setAddFlyoutVisibility}
+            loadActions={async () => {}}
+            editItem={() => {}}
+            isLoadingActions={false}
+            actions={[]}
+            setActions={() => {}}
+          />
+        </IntlProvider>
+      );
+      const createFirstActionButton = await screen.findByTestId('createFirstActionButton');
+      expect(createFirstActionButton).toBeEnabled();
+      userEvent.click(createFirstActionButton);
+      expect(setAddFlyoutVisibility).toBeCalled();
```

Comment:
nit:

```suggestion
      userEvent.click(createFirstActionButton);
      await waitFor(() => {
          expect(setAddFlyoutVisibility).toBeCalled();
      });
```

Two minor things that I would do differently based on the flaky tests we had in the past:
1. no need to check if the button is `enabled` when we are clicking it anyway in the next line.
2. better to wrap the `expect(function).toBeCalled()` in a `waitFor` block for a longer timeout.